### DISCUSSION TITLE
How to create form black with 7 consequtive select roll outs (pickers) for each column and show the result from last or first colum of database?

### DISCUSSION THREAD

**Discussion by FleetDriver** [Budibase/budibase#15035] [Help] [answered]
Hello,

I am new in this and would like to write some recommendation calculator where some of the data needs to be chosen.

We have csv file with 7 columns uploaded and we would like to create the form where the user can select the information from each column and at the end when all will be chosen after a click to "search" button it will trigger the result from first or last column in the same dataset.

I reached to the point where I could create form block from the datasource and it seems like earch column is separately show to be chosen, however nothing apprears there , I  cannot even open the roll out with select option.

<img width="953" alt="image" src="https://github.com/user-attachments/assets/21a89209-fec3-4e64-948f-2fafc9dd2f0c">

Also when I am trying the preview or publishing of website I receive the below 

<img width="896" alt="image" src="https://github.com/user-attachments/assets/eb0bcdf5-6431-4773-8759-949544ae564d">

I am totally new in this so  I am searching the possibilities.

At the end of the day I need to define also this search button to trigger the result from last or first column. 

If all of the selections will be done then there will be only one possibility for a result from first or last column.

Thanks upfront for the help!

**Comment by mikesealey** [comment, marked as answer]
Hey @FleetDriver 

A couple of things I've noticed from your screenshots:
You're fetching the row ID from the URL, but if you're previewing the screen with the form on, there's a high likelihood that you arrived at that URL without specifying the row ID in the URL. Typically we might do this by having a table of all of the rows on `/table` and then clicking a row might navigate to `/table/:rowID` 
Have a read through this guide in the documentation to make sure you're taking the right approach: https://docs.budibase.com/docs/passing-bindings-in-url-parameters

I can see that you've also got the form block set to _view_. If you want to use this form to edit, you'll still need to pass in a row ID, but it will allow the user to edit the row and save their changes.

With regards to your dropdowns - there are a few ways you could approach these - You've already started with a Form Block, so lets head down that route. If you click the little cog/sproket icon next to the field you'll be able to specify certain settings within each field. In your case, for select-pickers in the Form Block, you'll want to set the Options Source to "Custom". You will then see a button that says "Define Options" - here you can edit which options are presented to the user in the dropdown.

![image](https://github.com/user-attachments/assets/2601b37e-c717-4ae3-a596-94f743b2e5a1)

There are a few other ways to handle this kind of thing, using Multi-Select, Options Picker, or Relationship Picker, and whilst these options offer much greater control over the data, they require more setup. Head to https://docs.budibase.com/docs/pickers to read more about them.

You may also find this guide useful - https://docs.budibase.com/docs/cascading-dropdown-filters - Here we are filtering the options presented in the second dropdown, based on the value supplied in the first dropdown. You will only see cities that are in the country.

Hopefully this helps get you on your way. We're looking forward to seeing what you build with Budibase!

**Reply by FleetDriver** [reply]
Hey Mike,

Thanks a lot for your help, ineed seems like creating the new form(container) with option picker is better choice, however for relationship picker I am not able to choose the field

<img width="965" alt="image" src="https://github.com/user-attachments/assets/d4575b16-da2e-48f0-b412-251ddc811738">


Like it is written below:
<img width="532" alt="image" src="https://github.com/user-attachments/assets/7de89794-a7c3-47df-a6d5-f1a07dc44aed">


I even duplicated the database to provide for relationship picker the separate table (which I prefer not to have) and do not know how to enforce the different column from the same datasoruce table in the field place of relationship picker to apply the filtering.

Thanks for help upfront!

Best regards,
